### PR TITLE
Package destroy fixes

### PR DESCRIPTION
**Pull request by adrianschroeter**
and some small refactoring. webui test cases still missing unfortunatly

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -309,23 +308,21 @@ def check_write_access!(ignoreLock = nil)
...
   end
 
-  def check_devel_packages
-    if develpackages.any?
-      develpackages.each do |package|
-        # We only care about packages in foreign projects
-        if package.project.name != self.project.name
-          self.errors.add(:base, "used as devel package by #{package.project.name}/#{package.name}")
-        end
+  def check_weak_dependencies(ignore_local = false)
```
Comment (hennevogel) [inline src/api/app/models/package.rb]: I know you do love the future but nobody calls this with `ignore_local = true` so why add it?

**Comment by hennevogel** [comment]
I wonder why we don't force this anyway. In which circumstance is that changing my decision to delete my package?

**Comment by adrianschroeter** [comment]
On Dienstag, 27. Oktober 2015, 05:29:06 CET wrote Henne Vogelsang:
> I wonder why we don't force this anyway. In which circumstance is that changing my decision to delete my package?

Real life use case is that people get informed that this particular package instance
is actually used as devel area for openSUSE:Factory. Since there is the policy to
have one people should get a warning at least.

So you may delete also the openSUSE:Factory package or fix the devel definition there first.

File: src/api/app/models/package.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/app/models/package.rb]: hm, yes, my plan was to bring the Project.can_be_really_deleted? method in sync .... but we can as well skip it....

**Comment by hennevogel** [comment]
Okay, I've created #1308 to discuss this

**Pull request merged**